### PR TITLE
OSSM-3629 Add nightly image build jobs

### PR DESCRIPTION
**Pull request by dgn**

### CODE REVIEW

**Review by yxun — APPROVED** [review]

**Review by jwendell — APPROVED** [review]

File: ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-3.0.yaml [hunk 1]
```diff
@@ -114,6 +114,69 @@ tests:
...
+        # login to quay.io in the privileged pod
+        oc rsh -n "${MAISTRA_NAMESPACE}" "${MAISTRA_SC_POD}" \
+          docker login -u='maistra-dev+prow' \
+          -p=${QUAY_PASS} quay.io
+        # build and push images
+        oc rsh -n "${MAISTRA_NAMESPACE}" "${MAISTRA_SC_POD}" \
+          entrypoint \
+          sh -c ' \
+          export REPO=quay.io/maistra-dev/istio-ubi9-operator; \
+          export IMAGE=$REPO:3.0-latest; \
```
Comment (jwendell) [inline ci-operator/config/maistra/istio-operator/maistra-istio-operator-maistra-3.0.yaml:161]: nit: replace `;` with `&&` so if any command fail, execution stops

**Review by jwendell — APPROVED** [review]

**Pull request merged**